code/obj/item.dm

```diff
@@ -882,6 +906,47 @@
 		STOP_TRACKING_CAT(TR_CAT_BURNING_ITEMS)
 	burning_last_process = src.burning
 
+/obj/item/proc/process_corrode()
+	SHOULD_NOT_SLEEP(TRUE)
+	if (src.corrosion > 0)
+		var/acid_taken_mult = 1
+		if(src.hasProperty("acidres")) //multiplier from permeability coefficient
+			acid_taken_mult = 0.01 * (100 - src.getProperty("acidres"))
+		if (istype(src.material) && src.material.hasProperty("corrosion")) //multiplier from corrosion resistance
+			acid_taken_mult = acid_taken_mult * (0.01 * (100 - src.material.getProperty("corrosion")))
+
+		var/base_melt = max(0.1 * src.corrosion,1) //calculate how much corrosion to consume, with a simple floor.
+		src.corrosion -= base_melt //consume the corrosion accordingly.
+		src.health -= 0.1 * base_melt * acid_taken_mult //apply the mult at this phase, so more acid resistance doesn't just prolong damage
```

Comment:
back-of-the-envelope math suggests that, when kept constantly at maximum corrosion, it takes a solid 2 minutes to melt through a space helmet. Which kinda makes changeling spit pretty worthless

Reply:
Rectified.